manifests/init.pp

```diff
@@ -3,37 +3,53 @@
 # Manage facter
 #
 class facter (
-  Boolean $manage_facts_d_dir                   = true,
-  Boolean $purge_facts_d                        = false,
-  Stdlib::Absolutepath $facts_d_dir             = '/etc/facter/facts.d',
-  String $facts_d_owner                         = 'root',
-  String $facts_d_group                         = 'root',
-  Stdlib::Filemode $facts_d_mode                = '0755',
-  Stdlib::Absolutepath $path_to_facter          = '/opt/puppetlabs/bin/facter',
-  Stdlib::Absolutepath $path_to_facter_symlink  = '/usr/local/bin/facter',
-  Boolean $ensure_facter_symlink                = false,
-  Hash $facts_hash                              = {},
-  Boolean $facts_hash_hiera_merge               = false,
-  String $facts_file                            = 'facts.txt',
-  String $facts_file_owner                      = 'root',
-  String $facts_file_group                      = 'root',
-  Stdlib::Filemode $facts_file_mode             = '0644',
-  Stdlib::Absolutepath $facter_conf_dir         = '/etc/puppetlabs/facter',
-  String $facter_conf_dir_owner                 = 'root',
-  String $facter_conf_dir_group                 = 'root',
-  Stdlib::Filemode $facter_conf_dir_mode        = '0755',
-  String $facter_conf_name                      = 'facter.conf',
-  String $facter_conf_owner                     = 'root',
-  String $facter_conf_group                     = 'root',
-  Stdlib::Filemode $facter_conf_mode            = '0644',
-  Facter::Conf $facter_conf                     = {},
+  Boolean $manage_facts_d_dir,
```

Comment:
Data from common should stay as the defaults listed here. This makes it easier to understand what the defaults are as well as work with puppet-strings.